src/components/DirectorParamChange.tsx

```diff
@@ -0,0 +1,161 @@
+import type { Amount } from '@agoric/ertp/src/types';
+import clsx from 'clsx';
+import { motion } from 'framer-motion';
+import { LoadStatus, usePublishedDatum, WalletContext } from 'lib/wallet';
+import { useContext, useState } from 'react';
+import { AmountInput, PercentageInput } from './inputs';
+
+export type ParameterValue =
+  | {
+      type: 'invitation';
+      value: Amount<'set'>;
+    }
+  | {
+      type: 'ratio';
+      value: { numerator: Amount<'nat'>; denominator: Amount<'nat'> };
+    }
+  | {
+      type: 'amount';
+      value: Amount;
+    }
+  | {
+      type: 'string';
+      value: string;
+    };
+
+interface Props {
+  charterOfferId: number;
+}
+
+export default function DirectorParamChange(props: Props) {
+  const walletUtils = useContext(WalletContext);
+
+  const { data, status } = usePublishedDatum(`vaultFactory.governance`);
+  const [minutesUntilClose, setMinutesUntilClose] = useState(10);
+
+  const [paramPatch, setParamPatch] = useState({});
+
+  console.log('ProposeParamChange', { data, paramPatch });
+
+  const canMakeProposal = !!props.charterOfferId;
+
+  function displayParam(name: string, { type, value }: ParameterValue) {
+    console.log('display param', name, value, type);
+    switch (type) {
+      case 'amount':
+        return (
+          <AmountInput
+            suffix={name === 'MinInitialDebt' && 'IST'}
+            value={(paramPatch[name] || value).value}
```

Comment:
`value=value.value` :-|

`paramPatch[name] || value` is repeated in each case. please hoist above the switch. e.g. `const arg = paramPatch[name] || value`. Then this would be,
```suggestion
            value={arg.value}
```

Reply:
Done